tests/components/cast/test_init.py

```diff
@@ -137,27 +161,89 @@ def get_suggested(schema, key):
             return k.description["suggested_value"]
 
 
-async def test_option_flow(hass):
+@pytest.mark.parametrize(
+    "parameter_data",
+    [
+        (
+            "known_hosts",
+            ["192.168.0.10", "192.168.0.11"],
+            "192.168.0.10,192.168.0.11",
+            "192.168.0.1,  ,  192.168.0.2 ",
+            ["192.168.0.1", "192.168.0.2"],
+        ),
+        (
+            "uuid",
+            ["bla", "blu"],
+            "bla,blu",
+            "foo,  ,  bar ",
+            ["foo", "bar"],
+        ),
+        (
+            "ignore_cec",
+            ["cast1", "cast2"],
+            "cast1,cast2",
+            "other_cast,  ,  some_cast ",
+            ["other_cast", "some_cast"],
+        ),
+    ],
+)
+async def test_option_flow(hass, parameter_data):
     """Test config flow options."""
-    config_entry = MockConfigEntry(
-        domain="cast", data={"known_hosts": ["192.168.0.10", "192.168.0.11"]}
-    )
+    all_parameters = ["ignore_cec", "known_hosts", "uuid"]
+    parameter, initial, suggested, user_input, updated = parameter_data
+
+    data = {
+        "ignore_cec": [],
+        "known_hosts": [],
+        "uuid": [],
+    }
+    data[parameter] = initial
+    config_entry = MockConfigEntry(domain="cast", data=data)
     config_entry.add_to_hass(hass)
     await hass.async_block_till_done()
 
+    # Test ignore_cec and uuid options are hidden if advanced options are disabled
     result = await hass.config_entries.options.async_init(config_entry.entry_id)
```

Comment:
We need to set up the integration before starting the options flow to make sure the config flow module is loaded.

Reply:
Thanks a lot, Martin!
This is fixed by 7ea9378.

I'll open a separate PR to move the config flow tests to their own file.